### DISCUSSION TITLE
My graph doesn't draw the first point correctly

### DISCUSSION THREAD

**Discussion by ivevasiljevic** [patrykandpatrick/vico#613] [Questions] [answered]
So the data that feed into my graph are prices, and the first value of the array is 240.0. Seeing from the image below, the first point is presented as in it is around 238.5 approx. The threshold line is also added that shows that the first value from the array is indeed 240.0.

![Screenshot 2024-03-12 at 08 59 08](https://github.com/patrykandpatrick/vico/assets/30770600/2d3acb6a-9863-4227-a193-26ed90e65130)

If anyone encountered this problem and knows how to fix, it would be great :)

**Comment by ivevasiljevic** [comment]
From observing the data and extra store, I think that your libs being so depended on Float values, really is harming the preciseness and effectiveness. I have a huge need for a more precise type, such as Double. I am working with Epoch seconds and every minute and second is important to me.

I think what is going on is that I have some values calculated and since I have to transform them as Float, the precision gets lost and so does the correct mapping inside of the extra store.

Please bring support for Double, and/or other values.

Thank you :)

**Reply by patrickmichalik** [reply, marked as answer]
Hello! That’s most likely the cause of the issue. (To be sure, we’d need a minimal reproducible example.) We’ll be switching from `Float` to `Double` in Vico 2, but please note that `Float` is generally sufficient with the appropriate transformations. Here, I’d suggest subtracting the minimum timestamp from the _x_ values, storing it as an extra (`ExtraStore` supports all data types), and adding it back to the _x_ values at the formatting stage. The _x_ values will be significantly smaller, which should resolve any accuracy issues. The first whole number that `Float` can’t accurately represent is close to 17 million.
